File: src/main/java/de/cplaiz/activecraftcore/guis/profilemenu/ProfileMenu.java
package de.cplaiz.activecraftcore.guis.profilemenu;

import de.cplaiz.activecraftcore.ActiveCraftCore;
import de.cplaiz.activecraftcore.guicreator.GuiBackItem;
import de.cplaiz.activecraftcore.guicreator.GuiCloseItem;
import de.cplaiz.activecraftcore.guicreator.GuiCreatorDefaults;
import de.cplaiz.activecraftcore.guicreator.GuiPlayerHead;
import de.cplaiz.activecraftcore.guis.profilemenu.inventory.*;
import de.cplaiz.activecraftcore.messagesv2.ColorScheme;
import de.cplaiz.activecraftcore.messagesv2.MessageSupplier;
import de.cplaiz.activecraftcore.messagesv2.MessageSupplierKt;
import de.cplaiz.activecraftcore.playermanagement.Profilev2;
import lombok.Data;
import org.bukkit.entity.Player;

@Data
public class ProfileMenu {

    private Player player;
    private Player target;
    private Profilev2 profile;
    private GuiPlayerHead playerHead;
    private MainProfile mainProfile;
    private ActionProfile actionProfile;
    private ViolationsProfile violationsProfile;
    private GamemodeSwitcherProfile gamemodeSwitcherProfile;
    private HomeListPageLayout homeListPageLayout;
    private StorageProfile storageProfile;
    private final MessageSupplier messageSupplier;
    private final ColorScheme colorScheme;
    private GuiCloseItem defaultGuiCloseItem;
    private GuiBackItem defaultGuiBackItem;

    public ProfileMenu(Player player, Player target) {
        this.player = player;
        this.target = target;
        profile = Profilev2.of(target);
        this.messageSupplier = MessageSupplierKt.getMessageSupplier(player, ActiveCraftCore.getInstance().getActiveCraftMessagev2());
        this.colorScheme = messageSupplier.getColorScheme();
        defaultGuiCloseItem = new GuiCloseItem(GuiCreatorDefaults.closeItemDisplayname(messageSupplier));
        defaultGuiBackItem = new GuiBackItem(GuiCreatorDefaults.backItemDisplayname(messageSupplier));

        //playerhead
        playerHead = new GuiPlayerHead(target);
        playerHead.setLore(colorScheme.primary() + "aka " + profile.getNickname(), colorScheme.primaryAccent() + profile.getUuid().toString());
        playerHead.setDisplayName(colorScheme.primary() + target.getName());

        // menus
        mainProfile = new MainProfile(this);
        actionProfile = new ActionProfile(this);
        violationsProfile = new ViolationsProfile(this);
        gamemodeSwitcherProfile = new GamemodeSwitcherProfile(this);
        homeListPageLayout = new HomeListPageLayout(this);
        storageProfile = new StorageProfile(this);
    }
}

File: src/main/java/org/activecraft/activecraftcore/guis/profilemenu/ProfileMenu.kt
package org.activecraft.activecraftcore.guis.profilemenu

import org.activecraft.activecraftcore.ActiveCraftCore
import org.activecraft.activecraftcore.guicreator.GuiBackItem
import org.activecraft.activecraftcore.guicreator.GuiCloseItem
import org.activecraft.activecraftcore.guicreator.GuiCreatorDefaults.backItemDisplayname
import org.activecraft.activecraftcore.guicreator.GuiCreatorDefaults.closeItemDisplayname
import org.activecraft.activecraftcore.guicreator.GuiPlayerHead
import org.activecraft.activecraftcore.guis.profilemenu.inventory.*
import org.activecraft.activecraftcore.messages.ColorScheme
import org.activecraft.activecraftcore.messages.MessageSupplier
import org.activecraft.activecraftcore.messages.getMessageSupplier
import org.activecraft.activecraftcore.playermanagement.Profile
import org.activecraft.activecraftcore.playermanagement.Profile.Companion.of
import org.bukkit.entity.Player

class ProfileMenu(val player: Player, val target: Player) {
    val profile: Profile = of(target)
    val playerHead: GuiPlayerHead = GuiPlayerHead(target)
    val mainProfile: MainProfile
    val actionProfile: ActionProfile
    val violationsProfile: ViolationsProfile
    val gamemodeSwitcherProfile: GamemodeSwitcherProfile
    val homeListPageLayout: HomeListPageLayout
    val storageProfile: StorageProfile
    val messageSupplier: MessageSupplier = player.getMessageSupplier(ActiveCraftCore.INSTANCE.activeCraftMessage!!)
    val colorScheme: ColorScheme = messageSupplier.colorScheme
    val defaultGuiCloseItem: GuiCloseItem = GuiCloseItem(closeItemDisplayname(messageSupplier))
    val defaultGuiBackItem: GuiBackItem = GuiBackItem(backItemDisplayname(messageSupplier))

    init {
        //playerhead
        playerHead.setLore(
            colorScheme.primary.toString() + "aka " + profile.nickname,
            colorScheme.primaryAccent.toString() + profile.uuid.toString()
        )
        playerHead.setDisplayName(colorScheme.primary.toString() + target.name)

        // menus
        mainProfile = MainProfile(this)
        actionProfile = ActionProfile(this)
        violationsProfile = ViolationsProfile(this)
        gamemodeSwitcherProfile = GamemodeSwitcherProfile(this)
        homeListPageLayout = HomeListPageLayout(this)
        storageProfile = StorageProfile(this)
    }
}